### DISCUSSION TITLE
dumpdata with django-scopes

### DISCUSSION THREAD

**Discussion by AliveDevil** [TandoorRecipes/recipes#1832] [Q&A] [answered]
No python/django user here, though I'd like to transfer all data from SQLite (for testing purposes) to postgreSql. Using an SQL dump doesn't look that compelling, so I though that `dumpdata` would be more suitable. Turns out that django-scopes doesn't like django-admin, which results in this output:
```
./manage.py dumpdata cookbook --natural-foreign --natural-
primary
[{"model": "cookbook.foodinheritfield", "pk": 1, "fields": {"field": "supermarket_category", "name": "Supermarket Category"}}, {"model": "cookbook.foodinheritfield", "pk": 2, "fields": {"field": "ignore_shopping", "name": "Ignore Shopping"}}, {"model": "cookbook.foodinheritfield", "pk": 3, "fields": {"field": "diet", "name": "Diet"}}, {"model": "cookbook.foodinheritfield", "pk": 4, "fields": {"field": "substitute", "name": "Substitute"}}, {"model": "cookbook.foodinheritfield", "pk": 5, "fields": {"field": "substitute_children", "name": "Substitute Children"}}, {"model": "cookbook.foodinheritfield", "pk": 6, "fields": {"field": "substitute_siblings", "name": "Substitute Siblings"}}, {"model": "cookbook.space", "pk": 1, "fields": {"name": "Default", "created_by": ["admin"], "created_at": "2022-05-20T22:30:37.074Z", "message": "", "max_recipes": 0, "max_file_storage_mb": 0, "max_users": 0, "allow_sharing": true, "demo": false, "show_facet_count": false, "food_inherit": []}}CommandError: Unable to serialize database: A scope on dimension(s) space needs to be active for this query
```

This is acknowledged by the django-scopes developers[^1], but I don't see how the entry code snippet can be integrated to get the raw json output with disabled scopes.
This is broken/incompatible since c42266b8.

What do I need to edit in order to get the cookbook data dumps working correctly?

[^1]: https://github.com/raphaelm/django-scopes/issues/13

**Comment by smilerz** [comment]
Have you tried exporting the recipes (/export/) and attempting to import them again?

**Reply by AliveDevil** [reply]
Didn't see that. Thanks for the pointer, unfortunately it doesn't export references to cook books, which I'd need to recreate.

**Comment by smilerz** [comment]
It’s definitely not perfect.  If you create a feature request that includes the missing components that you miss most it’ll get included eventually. (Which won’t help you today, sadly)

**Comment by vabene1111** [comment, marked as answer]
sorry but i dont know how to use dumpdata with scopes, i guess there is a way of invocing the dump command within scopes disabled but i am not sure how and also sadly dont have the time to properly look into this. 

I am sorry but i just cant find/justify the time to implement some custom solution here, we will at some point offer backups over the webinterface but thats not done yet. 

Maybe this can help you https://pgloader.readthedocs.io/en/latest/ref/sqlite.html

**Reply by AliveDevil** [reply]
For me this[^1] looks like a viable solution.
Implements import and export as separate commands, which extend loaddata and dumpdata, disabling scopes before going into super implementation (just like rebuildindex does).

[^1]: https://github.com/TandoorRecipes/recipes/pull/1854
